Makefile

```diff
@@ -0,0 +1,87 @@
+.PHONY: deps
```

Comment:
why are all those `.PHONY` needed?

Reply:
`.PHONY` needed for make to ignore any files names same as the target. If you happen have a file name `deps`, make will basically ignore `make deps`

Reply:
Right but we don't have a `deps` directory do we?

Reply:
Right now we don't, but if would have it by any chance in future, you'll have a good time scratching your head. @BackEndTea did it all right with `.PHONY`

Reply:
That's how they do it in the make world. Can't help this.

Reply:
It's not a requirement here so it's about what we prefer.

Reply:
Also see https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html:

> There are two reasons to use a phony target: to avoid a conflict with a file of the same name, and to improve performance.

Reply:
@localheinz it's up to you, personally I would hate to always have to add a target to the .PHONY (and perf. is very unlikely to ever be an issue it's a micro-optim to say the least here) but whatever ways you guy swing I don't feel strongly about it